ultralytics/data/converter.py

```diff
@@ -474,3 +474,124 @@ def merge_multi_segment(segments):
                     nidx = abs(idx[1] - idx[0])
                     s.append(segments[i][nidx:])
     return s
+
+
+def yolo_bbox2obb(
+    data,
+    obb_save,
+    seg_save=None,
+    segment_data=False,
+    im_ext=[".jpg", ".jpeg"],
+    det_model="yolov8x.pt",
+    sam_model="sam_b.pt",
+) -> None:
+    """
+    Converts existing object detection dataset (bounding boxes) or segmentation dataset in YOLO format into YOLO
+    oriented bounding box (OBB) format. Generates intermediate segmentation data using SAM auto-annotator as needed.
+
+    Args:
+        data (str|Path): Path to data to convert, either object detection (vertical bounding box) or segmentation data.
+                        Expected to have all images and labels in the same directory.
+        obb_save (str|Path): Path to save the output OBB annotation files
+        seg_save (str|Path): Path to save generated segmentation data; optional.
+        segment_data (bool): Indicates if the `data` path contains segmentation data; optional.
+        im_ext (list[str]): List of image file extensions to fetch from the `data` path; optional.
+        det_model (str): Detection model to use for generating segmentation data, use custom trained model for custom data; optional.
+        sam_model (str): Segmentation model to use for intermediate segmentation data; optional.
+
+    Notes:
+        The input directory structure assumed for dataset:
+            - data
+                ├─ 001.jpg
```

Comment:
I agree with all of these and could go either way with using a single `save_dir` or just placing everything in the `data` directory. Would it make sense to do something like this so users wouldn't have to move files around?

```
└── data
    ├── images
    └── labels
            └── detect
            └── segment
            └── obb
```

Reply:
@Burhan-Q @Laughing-q ah this may not work with our default dataset format as the `/images/` and `/labels/` directories need to be next to each other to find the labels from the images.

We probably want to do this if no `save_dir` is not passed, and then allow `save_dir` to override for a custom save directory.

```
└── data
    ├── images
    ├── labels
    ├── labels-segment(newly generated)
    ├── labels-obb(newly generated)
```